packages/functional-tests/pages/products/index.ts

```diff
@@ -7,16 +9,50 @@ export class SubscribePage extends BaseLayout {
 
   async setCreditCardInfo() {
     const frame = this.page.frame({ url: /elements-inner-card/ });
+    await frame.fill('.InputElement[name=cardnumber]', '');
     await frame.fill('.InputElement[name=cardnumber]', '4242424242424242');
     await frame.fill('.InputElement[name=exp-date]', '555');
     await frame.fill('.InputElement[name=cvc]', '333');
     await frame.fill('.InputElement[name=postal]', '66666');
     await this.page.check('input[type=checkbox]');
+    await this.page.click('button[type=submit]');
+  }
+
+  async setFailedCreditCardInfo() {
+    const frame = this.page.frame({ url: /elements-inner-card/ });
+    await frame.fill('.InputElement[name=cardnumber]', '4000000000000341');
+    await frame.fill('.InputElement[name=exp-date]', '666');
+    await frame.fill('.InputElement[name=cvc]', '444');
+    await frame.fill('.InputElement[name=postal]', '77777');
+    await this.page.check('input[type=checkbox]');
+    await this.page.click('button[type=submit]');
+  }
+
+  async setPayPalInfo() {
+    await this.page.check('[data-testid="confirm"]');
+    const [paypalWindow] = await Promise.all([
+      this.page.waitForEvent('popup'),
+      this.page.click('[data-testid="paypal-button-container"]'),
+    ]);
+    await paypalWindow.waitForLoadState('load');
+    await paypalWindow.fill(
+      'input[type=email]',
+      'qa-test-no-balance-16@personal.example.com'
+    );
+    await paypalWindow.click('button[id=btnNext]');
+    await paypalWindow.waitForLoadState('load');
+    await paypalWindow.fill('input[type=password]', 'Ah4SboP6UDZx95I');
+    await paypalWindow.click('button[id=btnLogin]');
+    await paypalWindow.click('button[id=consentButton]');
+  }
+
+  async clickTryAgain() {
+    return Promise.all([this.page.click('[data-testid="retry-link"]')]);
```

Comment:
nit: not sure if this needs a `Promise.all()` wrapper if it's the only promise in an array.

Maybe just this?
```js
return this.page.click('[data-testid="retry-link"]');
```